### PR TITLE
[PM-5276] Migrate FolderService to state providers

### PR DESCRIPTION
**Pull request by gbubemismith**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/browser/src/popup/services/services.module.ts [hunk 1]
```diff
@@ -207,20 +207,33 @@ function getBgService<T>(service: keyof MainBackground) {
...
       },
-      deps: [CryptoService, I18nServiceAbstraction, CipherService, StateServiceAbstraction],
+      deps: [
+        CryptoService,
+        I18nServiceAbstraction,
+        CipherService,
+        StateServiceAbstraction,
+        StateProvider,
+      ],
     },
```
Comment (justindbaur) [inline apps/browser/src/popup/services/services.module.ts:233]: You should be able to delete this and rely on the one supplied by `jslib-services.module.ts`.

**Review by justindbaur — CHANGES_REQUESTED** [review]
You should also be able to delete `{set|get}{Encrypted|Decrypted}Folders` methods from state service and the associated properties off of the account object.

**Review by LRNcardozoWDF — DISMISSED** [review]
LGTM!

File: libs/common/src/platform/state/state-definitions.ts [hunk 2]
```diff
@@ -26,3 +26,5 @@ export const ENVIRONMENT_DISK = new StateDefinition("environment", "disk");
 
 export const GENERATOR_DISK = new StateDefinition("generator", "disk");
 export const GENERATOR_MEMORY = new StateDefinition("generator", "memory");
+
+export const FOLDER_DISK = new StateDefinition("folder", "disk");
```
Comment (justindbaur) [inline libs/common/src/platform/state/state-definitions.ts:30]: I should have noticed this on my first review sorry, but I believe you want the `{ web: "memory" }` override. Like @shane-melton did and I talked about [here](https://github.com/bitwarden/clients/pull/7680/files#r1467776755). 

Along with doing this you have some more methods you can delete from the web override of `StateService`.

**Review by justindbaur — CHANGES_REQUESTED** [review]
Something I should have caught the first time but super close.

**Review by justindbaur — APPROVED** [review]
Looks great, thanks!

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Pull request merged**